pom.xml

```diff
@@ -4,14 +4,18 @@
   <parent>
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>1.532.3</version>
+    <version>2.31-SNAPSHOT</version>
   </parent>
   <artifactId>validating-string-parameter</artifactId>
   <packaging>hpi</packaging>
   <name>Validating String Parameter Plugin</name>
-  <version>2.4-SNAPSHOT</version>
+  <version>2.5-SNAPSHOT</version>

```

Comment:
My reasoning was that in keeping with semantic versioning, the minor number should be increased: this is a feature of the plugin that did not exist previously, but does not introduce backwards-incompatible changes. Thus, it doesn't warrant a major version bump, but certainly warrants one.

Since there are no clear-cut contributor guides, I'll go with what you decide is convention.